shared/lib/formatting/index.js

```diff
@@ -332,8 +335,37 @@ function unescapeAndFormatSpaces( str ) {
 	return decodeEntities( str ).replace( / /g, nbsp );
 }
 
+/**
+ * Traverse an object or array decoding HTML entities on string properties
+ * Accepts a paths argument selecting which properties to update (See lodash.get)
+ *
+ * It does its job in place, so the object passed will be updated.
+ * This helper is used to decode HTML entities in strings properties,
+ * on an object.
+ * We need to decode HTML entities because the REST API returns them
+ * already encoded, and React will encode them again.
+ *
+ * @param	{Object} obj Object to traverse in preparation for React rendering
+ * @param {String/Array} paths Path or paths to decode.
+ */
+function traverseAndDecodeEntities( obj, paths ) {
```

Comment:
In this usage, I agree that there's likely not a downside. But, you're also attempting to add a method to `lib/formatting` which is used throughout the project, so it is likely that the method would not be used in the same manner.

You could do something like the following which doesn't mutate data and doesn't require you to map the entire object.

```
let decodedValues = {
	display_name: decodeEntities( data.display_name )
	user_URL: decodeEntities( data.user_URL )
	description: decodeEntities( data.description )
};

this.settings = Object.assign( {}, data, decodedValues );
```